spec/requests/categories_controller_spec.rb

```diff
@@ -139,6 +139,22 @@
       expect(subsubcategory_response["topics"].map { |c| c['id'] }).to contain_exactly(topic3.id)
     end
 
+    it 'includes topics filtered by tag for categories when requested' do
+      tag = Fabricate(:tag, name: "test-tag")
+      category = Fabricate(:category)
+      topic = Fabricate(:topic, category: category, tags: [tag])
+      topic2 = Fabricate(:topic, category: category)
+      CategoryFeaturedTopic.feature_topics
```

Comment:
Can we test a few more cases? Like

- create a few more topics with other tags and ensure they are not listed
- create a category and a subcategory and a post with a tag in the subcategory and ensure it is listed